app/actions/loadIcoParams.ts

```diff
@@ -0,0 +1,32 @@
+import { BigNumber } from "bignumber.js";
+import * as moment from "moment";
+import { Dispatch } from "redux";
+import { selectAddress } from "../reducers/icoParameters";
+import { loadIcoParamsFromEnviroment } from "../utils";
+import loadIcoParamsFromContract from "../web3/loadIcoParamsFromContract";
+import { LOAD_ICO_PARAMS } from "./constants";
+
+export function loadIcoParamsAction(
+  startDate: string,
+  endDate: string,
+  minCap: BigNumber,
+  maxCap: BigNumber
+) {
+  return {
+    type: LOAD_ICO_PARAMS,
+    payload: {
+      startDate,
+      endDate,
+      minCap,
+      maxCap,
+    },
+  };
+}
+
+export async function loadIcoParams(dispatch: Dispatch<any>, getState: any) {
+  const address = selectAddress(getState().icoParameters);
+  // TODO :Choose when to load from ENV or Smart_Contract
+  // const { minCap, maxCap, startDate, endDate } = await loadIcoParamsFromContract(address); // FROM SMART CONTRACT
+  const { minCap, maxCap, startDate, endDate } = await loadIcoParamsFromEnviroment(); // FROM ENV
+  dispatch(loadIcoParamsAction(startDate, endDate, minCap, maxCap));
```

Comment:
I would assume that we can go to way's: 
- Create two different states for pre ICO 
1- with ENV variables, where it would be limited and the only important prop is startDate
2- with smart-contracts, where it would throw if there is anything wrong with connecting with the rpc provider or address.
Or one pre-ICO state that handels both cases and checks